testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/OIDCClientRegistrationTest.java

```diff
@@ -468,5 +470,50 @@ private ClientRepresentation getKeycloakClient(String clientId) {
         return ApiUtil.findClientByClientId(adminClient.realms().realm(REALM_NAME), clientId).toRepresentation();
     }
 
+    @Test
+    public void testClientWithScope() throws Exception {
+        OIDCClientRepresentation clientRep = null;
+        OIDCClientRepresentation response = null;
+        String clientScope = "phone address";
+
+        clientRep = createRep();
+        clientRep.setScope(clientScope);
+        response = reg.oidc().create(clientRep);
+
+        Set<String> clientScopes = new HashSet<>(Arrays.asList(clientScope.split(" ")));
+        Set<String> registeredClientScopes = new HashSet<>(Arrays.asList(response.getScope().split(" ")));
+        assertTrue(clientScopes.equals(registeredClientScopes));
+    }
 
+    @Test
+    public void testClientWithNotDefinedScope() throws Exception {
+        OIDCClientRepresentation clientRep = null;
+        OIDCClientRepresentation response = null;
+
+        String clientScope = "notdefinedscope address";
+
+        clientRep = createRep();
+        clientRep.setScope(clientScope);
+        try {
+            response = reg.oidc().create(clientRep);
+            fail("Expected 403");
+        } catch (ClientRegistrationException e) {
+            assertEquals(403, ((HttpErrorException) e.getCause()).getStatusLine().getStatusCode());
+        }
+    }
+
+    @Test
+    public void testClientWithoutScope() throws ClientRegistrationException {
+        Set<String> realmOptionalClientScopes = new HashSet<>(adminClient.realm(REALM_NAME).getDefaultOptionalClientScopes()
+                .stream().map(i->i.getName()).collect(Collectors.toList()));
+
+        OIDCClientRepresentation clientRep = null;
+        OIDCClientRepresentation response = null;
+
+        clientRep = createRep();
+        response = reg.oidc().create(clientRep);
+
+        Set<String> registeredClientScopes = new HashSet<>(Arrays.asList(response.getScope().split(" ")));
+        assertTrue(realmOptionalClientScopes.equals(new HashSet<>(registeredClientScopes)));
```

Comment:
Same as above. Is it possible to obtain client through admin REST API and doublecheck that default client scopes are realm default client scopes?

Reply:
@mposolda I see. I'll add the test as you suggested.